lib/matplotlib/streamplot.py

```diff
@@ -176,18 +176,14 @@ def streamplot(axes, x, y, u, v, density=1, linewidth=None, color=None,
     if use_multicolor_lines:
         if norm is None:
             norm = mcolors.Normalize(color.min(), color.max())
-        if cmap is None:
-            cmap = cm.get_cmap(matplotlib.rcParams['image.cmap'])
-        else:
-            cmap = cm.get_cmap(cmap)
+        cmap = cm.get_cmap(cmap)
 
     streamlines = []
     arrows = []
     for t in trajectories:
-        tgx = np.array(t[0])
-        tgy = np.array(t[1])
+        tgx, tgy = t.T
```

Comment:
The above seemed overly paranoid about making things arrays, but are we sure `t` is an arrray at this point?

Reply:
Yes, because it is filled with the return value of `integrate()`, which I changed below to return arrays (or None, which are filtered out).